modules/tinymce/src/plugins/table/main/ts/ui/UiUtils.ts

```diff
@@ -24,7 +29,16 @@ const onSetupToggle = (editor: Editor, formatName: string, formatValue: string)
       const setActive = (matched: boolean) =>
         api.setActive(isNone ? !matched : matched);
 
-      setActive(editor.formatter.match(formatName, { value: formatValue }, undefined, isNone));
+      const selectedCells = TableSelection.getCellsFromSelection(Util.getSelectionStart(editor), selections, Util.getIsRoot(editor));
+
+      const checkNode = (cell: SugarElement<Element>) =>
+        editor.formatter.match(formatName, { value: formatValue }, cell.dom, isNone);
+
+      if (isNone) {
+        setActive(Arr.exists(selectedCells, checkNode));
```

Comment:
This logic is confusing as it's doing a double negation that's not immediately obvious so can we please rework it and add some comments? e.g.:
```
// For the None-entry we need to check that no format matches, whereas for others we need every cell to match the format
const predicate = isNone ? Arr.exists : Arr.forall;
const matches = predicate(selectedCells, (cell) => editor.formatter.match(formatName, { value: formatValue }, cell.dom, isNone));
setActive(matches);
```

Another option is to move the `setActive` function closer since it has a comment already.

Reply:
Moved setActive closer.

Reply:
That only solves the comment part, not the confusing double negation I mentioned. Additionally, the if branches should always call the same thing, so having them operate separately will likely lead to maintenance issues in the future.

Reply:
Made some changes, not sure the comment applies as much anymore.

Reply:
This goes the opposite direction I wanted... it's now removed the `setActive` function which was shared code with the commented out `formatChanged` logic. It also doesn't solve the concern of having 2 branches that are largely trying to do the same thing.

Reply:
On second thoughts this is probably nitpicking a little too much. My point is still accurate, but it's not likely to happen when both are side-by-side. There is still the issue of the logic being inlined that was shared code, but we can cross that bridge when we get to fixing the other issue.